From 67740db2a9813d0d747a2c203bc7403692b34334 Mon Sep 17 00:00:00 2001
From: Yang Bo <yangbo@deepin.com>
Date: Thu, 29 Sep 2016 18:08:11 +0800
Subject: [PATCH] EDID reading failure in valleyview(device id:0x0f31)

EDID reading failure is observed in valleyview(device id: 0x0f31)
when using DVI-to-VGA connector. It is safe to use IS_VALLEYVIEW(dev)
here since successful EDID reading will goto out because ret != 0.
Only failure valleyview cards probe digital port.

Signed-off-by: Yang Bo <yangbo@deepin.com>
---
 drivers/gpu/drm/i915/intel_crt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_crt.c b/drivers/gpu/drm/i915/intel_crt.c
index 827b6ef..1571cd7 100644
--- a/drivers/gpu/drm/i915/intel_crt.c
+++ b/drivers/gpu/drm/i915/intel_crt.c
@@ -715,7 +715,7 @@ static int intel_crt_get_modes(struct drm_connector *connector)
 
 	i2c = intel_gmbus_get_adapter(dev_priv, dev_priv->vbt.crt_ddc_pin);
 	ret = intel_crt_ddc_get_modes(connector, i2c);
-	if (ret || !IS_G4X(dev))
+	if (ret || !(IS_G4X(dev) || IS_VALLEYVIEW(dev)))
 		goto out;
 
 	/* Try to probe digital port for output in DVI-I -> VGA mode. */
-- 
2.9.3

